### PR TITLE
Fix and omptimize flip API

### PR DESCRIPTION
**Pull request by sljlp**
### PR types

Function optimization
### PR changes

APIs
### Describe

1. add a feature supporting axis's type to be int in paddle.flip
2. use core.ops.flip in dygraph mode
3. update English doc for paddle.flip

### CODE REVIEW

File: python/paddle/tensor/manipulation.py [hunk 1]
```diff
@@ -240,10 +240,17 @@ def flip(x, axis, name=None):
           x = np.arange(image_shape[0] * image_shape[1] * image_shape[2]).reshape(image_shape)
           x = x.astype('float32')
           img = paddle.to_tensor(x)
-          out = paddle.flip(img, [0,1])
+          tmp = paddle.flip(img, [0,1])
+          print(tmp) # [[[10,11][8, 9]],[[6, 7],[4, 5]] [[2, 3],[0, 1]]]
```
Comment (wangxicoding) [inline python/paddle/tensor/manipulation.py]: 结果里面是不是有逗号少了

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -240,10 +240,17 @@ def flip(x, axis, name=None):
...
 
-          print(out) # [[[10,11][8, 9]],[[6, 7],[4, 5]] [[2, 3],[0, 1]]]
+          out = paddle.flip(tmp,-1)
+          print(out) # [[[11,10][9, 8]],[[7, 6],[5, 4]] [[3, 2],[1, 0]]]
```
Comment (wangxicoding) [inline python/paddle/tensor/manipulation.py]: 同上

Comment (sljlp) [reply python/paddle/tensor/manipulation.py]: done

**Review by wangxicoding — APPROVED** [review]
LGTM

**Review by PangHua — APPROVED** [review]

**Pull request merged**